### ISSUE TITLE
Delete Trick Number Parameter From Agent Action

### ISSUE THREAD

**Issue by aphelps4** [c-to-the-fazzy/cardiathena#55]
# Description

agent_turn was changed to accept the trick number in order to determine what actions should be made. We should use the passed in partial_state to calculate that trick number instead so agent_turn can remain generic. Simultaneous pass should be merged into master before starting this issue.

# Tasks

Include checkboxes for the sub tasks that need to be accomplished in order to mark the Development Task is complete e.g.

- [x] merge simultaneous pass pull request
  - [x] review code
- [x] get rid of trick_number from agent_turn
  - [x] use pass_type to determine if passing
  - [x] replace calls of parameter trick_number passing checks
  - [x] get rid of trick_number from parameters
